avocado/utils/configure_network.py

```diff
@@ -201,3 +202,18 @@ def is_interface_link_up(interface):
         log.info("Interface %s link is up", interface)
         return True
     return False
+
+
+def get_interface_ip(interface):
+    """
+    Get the IP address from a network interface
+    :param interface: name of the interface
+    :return: List of IP addresses.
+    """
+    cmd = "ip addr show %s" % interface
+    for line in process.system_output(cmd, shell=True).decode("utf-8") \
```

Comment:
You don't really need `shell=True` here so it'd be better (and especially safer) to not to use it (to avoid people passing interface name `eth0 ; rm -Rf / --no-preserve-root`)

Reply:
and do you actually need to decode? It'd be probably better to only decode the matched string ...

Reply:
@ldoktor thanks for a review:)
Yes, it makes sense to remove shell=True

Regarding decode, but cmd outputs is a bytes-like object and I can't use
a string pattern on it

Reply:
What about a different approach? `ip` command has support for `-4` (ipv4), `-6` (ipv6) and `-j` (json) format! :)

```python
cmd = "ip -4 -j address show lo"
output = json.loads(s.cmd(cmd).stdout)
print(output[0]['addr_info'][0]['local'])
'127.0.0.1' 
```

What do you think?

Reply:
Also, keep in mind that there is another PR (#3569 ) in the review. I think that we should join the efforts here.

Reply:
With regards to using `-j`, I've had similar experience with another tool (IIRC multipath). It's worth pointing out that not much older versions of multipath used to have no JSON output support.  I've tested this with iproute 4.1 on an EL 7 machine, and it has no support for `-j`.  Unfortunately, I've also found that `ip -V` gives very bad version numbers to use.

Reply:
Hum... :(

Yep, looks like was introduced on 4.17.

Reply:
@anascko you can't use string patterns, but you can use bytes patterns... `br"my_cool_bytes_pattern"`